src/util/UniqueCleanup.h

```diff
@@ -47,6 +47,10 @@ class UniqueCleanup {
 
   UniqueCleanup& operator=(UniqueCleanup&& cleanupDeleter) noexcept = default;
 
+  /// Runs the cleanup call preemptively by moving out of this instance and
+  /// immediately destructing the new instance
+  void invokeManuallyAndCancel() { auto self = std::move(*this); }
```

Comment:
This function should either be `&&` qualified, or not move the actual value, else that is really unexpected.
I suggest leaving the value as is and only calling the cleanup and setting `active_` to false.